libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -508,6 +509,29 @@ impl<'a> Validator<'a> {
                     ));
                 }
 
+                let referential_integrity = self.source.map(|ds| ds.referential_integrity()).unwrap_or_default();
+
+                if !referential_integrity.uses_foreign_keys()
+                    && rel_info
+                        .on_update
+                        .map(|act| act != ReferentialAction::NoAction)
+                        .unwrap_or(false)
+                {
+                    let ast_field = ast_model.find_field_bang(&field.name);
+                    let message =
+                        "Referential actions other than `NoAction` will not work for `onUpdate` without foreign keys. Please follow the issue: https://github.com/prisma/prisma/issues/9014";
+
+                    let span = ast_field
+                        .span_for_argument("relation", "onUpdate")
+                        .unwrap_or(ast_field.span);
+
+                    errors.push_error(DatamodelError::new_attribute_validation_error(
+                        message,
+                        RELATION_ATTRIBUTE_NAME,
+                        span,
+                    ));
+                }
```

Comment:
I think this should already be possible to validate this in `db/attributes/relation.rs` (as an exercise if you feel like it, this looks good and we can merge :+1:)

Reply:
Sounds fun, let me see how much work this would be, if it's doable before lunch I'll try, otherwise I need to finish the docs effort.